From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 22 Mar 2019 03:30:10 +0000
Subject: powerpc: vdso: Make vdso32 installation conditional in vdso_install
Bug-Debian: https://bugs.debian.org/785065
Forwarded: https://lore.kernel.org/linuxppc-dev/20190322042436.nttfgsdpdshco27y@decadent.org.uk/

The 32-bit vDSO is not needed and not normally built for 64-bit
little-endian configurations.  However, the vdso_install target still
builds and installs it.  Add the same config condition as is normally
used for the build.

Fixes: e0d005916994 ("powerpc/vdso: Disable building the 32-bit VDSO ...")
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 arch/powerpc/Makefile | 2 ++
 1 file changed, 2 insertions(+)

--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -403,7 +403,9 @@ vdso_install:
 ifdef CONFIG_PPC64
 	$(Q)$(MAKE) $(build)=arch/$(ARCH)/kernel/vdso64 $@
 endif
+ifdef CONFIG_VDSO32
 	$(Q)$(MAKE) $(build)=arch/$(ARCH)/kernel/vdso32 $@
+endif
 
 archclean:
 	$(Q)$(MAKE) $(clean)=$(boot)
